### PR TITLE
NUnit 3.6.1 update

### PR DESCRIPTION
**Pull request by BurningEnlightenment**
The update is necessary, because nunit 2 isn't supported on .net core. Blocks #155 

The changes necessary for the NUnit 3 update touch that many lines that I want them to be reviewed seperately. So please feel free to review them here.
For details see the commit message.

### CODE REVIEW

**Comment by BurningEnlightenment** [comment]
finally squashed after I won the fight against travis 🙈

File: libsodium-net/PublicKeyBox.cs [hunk 1]
```diff
@@ -81,7 +81,7 @@ public static byte[] Create(byte[] message, byte[] nonce, byte[] secretKey, byte
 
       //validate the length of the public key
       if (publicKey == null || publicKey.Length != PublicKeyBytes)
-        throw new KeyOutOfRangeException("publicKey", (publicKey == null) ? 0 : secretKey.Length,
+        throw new KeyOutOfRangeException("publicKey", (publicKey == null) ? 0 : publicKey.Length,
```
Comment (bitbeans) [inline libsodium-net/PublicKeyBox.cs]: good catch!

File: Tests/GenericHashExceptionTest.cs [hunk 2]
```diff
@@ -9,147 +9,164 @@ namespace Tests
...
+      Assert.Throws<KeyOutOfRangeException>(() => GenericHash.Hash("Adam Caudill", KEY, BYTES));
     }
 
     [Test]
-    [ExpectedException(typeof(BytesOutOfRangeException))]
     public void GenericHashBytesTooLong()
     {
       const string KEY = "1234567891123456";
       const int BYTES = 128;
-      GenericHash.Hash("Adam Caudill", KEY, BYTES);
```
Comment (BurningEnlightenment) [inline Tests/GenericHashExceptionTest.cs]: I killed this test 🙈; needs fix

Comment (BurningEnlightenment) [inline Tests/GenericHashExceptionTest.cs]: fixed

**Pull request merged**